### DISCUSSION TITLE
Is there a way to trim values, with the same implementation as .replace()?

### DISCUSSION THREAD

**Discussion by FonixAU** [dbt-labs/dbt-core#10821] [Q&A] [answered]
I am trying to do some dynamic flattening, and the only issue I cant get around, is that there are some characters I am replacing, that may appear at the ends and inside the string, so I am not able to stop my macro from creating trailing underscores

eg:`{{ row.replace("．",'_')
                                 .replace(".",'_')
                                 .replace(" : ",'_')
                                 .replace(":",'')
                                 .replace(",",'_')
                                 .replace("?",'')
                                 .replace("-", '_')
                                 .replace('/', '_')
                                 .replace('(','_')
                                 .replace(')','')
                                 .replace(' ', '_')
                                 .replace('___','_')
                                 .replace('__','_').upper() }}`

**Comment by jpgomezv** [comment, marked as answer]
Jinja is Python under the hood, so you can use Python string methods directly on the value. Add `.rstrip('_')` at the end of your chain — it removes any number of trailing underscores, which is exactly what your `replace(...)` calls leave behind:

{{ row.replace("．",'_')
        .replace(".",'_')
        .replace(" : ",'_')
        .replace(":",'')
        .replace(",",'_')
        .replace("?",'')
        .replace("-", '_')
        .replace('/', '_')
        .replace('(','_')
        .replace(')','')
        .replace(' ', '_')
        .replace('___','_')
        .replace('__','_')
        .upper()
        .rstrip('_') }}

If leading underscores can also appear (e.g. the value starts with a replaced character), add `.lstrip('_')` as well. Note that `rstrip`/`lstrip` operate on the string as a whole, so they're safe to use after collapsing runs with `.replace('___','_')` / `.replace('__','_')`.

If this is for generating column or alias names, an even cleaner option is the `dbt_utils.slugify` macro, which lowercases and collapses runs of non-alphanumeric characters into single underscores.

**Comment by jpgomezv** [comment]
Just a friendly follow-up — if my earlier answer helped resolve your issue, feel free to mark it as the accepted answer. Thanks!

**Comment by FonixAU** [comment]
Hi I havent worked in that team in over 2 years but I will let them know.
Thank you.
